trade-finance-manager-ui/component-tests/utilisation-reports/utilisation-report-reconciliation-for-report.component-test.js

```diff
@@ -84,4 +87,59 @@ describe(page, () => {
 
     wrapper.expectElement(`${premiumPaymentsTabSelector} table[data-cy="premium-payments-table"]`).toExist();
   });
+
+  it('should not render add payment button for PDC_READ user', () => {
+    const user = {
+      ...aTfmSessionUser(),
+      teams: [TEAM_IDS.PDC_READ],
+    };
+
+    wrapper = render({
+      ...params,
+      user,
+    });
+
+    const premiumPaymentsTabSelector = 'div#premium-payments';
+    wrapper.expectElement(`${premiumPaymentsTabSelector} input[data-cy="add-a-payment-button"]`).notToExist();
+  });
+
+  it('should not render generate keying data button for PDC_READ user', () => {
+    const user = {
+      ...aTfmSessionUser(),
+      teams: [TEAM_IDS.PDC_READ],
+    };
+
+    wrapper = render({
+      ...params,
+      user,
+    });
+
+    const premiumPaymentsTabSelector = 'div#premium-payments';
+    wrapper.expectElement(`${premiumPaymentsTabSelector} input[data-cy="generate-keying-data-button"]`).notToExist();
+  });
+
+  it('should pass userCanEdit as true to the premium payments table macro for a PDC_RECONCILE user', () => {
+    const user = aTfmSessionUser();
+
+    wrapper = render({
+      ...params,
+      user,
+    });
+
+    wrapper.expectElement(`div#premium-payments input[type="checkbox"]`).toExist();
+  });
+
+  it('should pass userCanEdit as false to the premium payments table macro for a PDC_READ user', () => {
+    const user = {
+      ...aTfmSessionUser(),
+      teams: [TEAM_IDS.PDC_READ],
+    };
+
+    wrapper = render({
+      ...params,
+      user,
+    });
+
+    wrapper.expectElement(`div#premium-payments input[type="checkbox"]`).notToExist();
+  });
```

Comment:
Thanks for adding these! 

The test names don't quite match the assertions. You and I both know that that is why this element does or doesn't exist and that that is why we are adding the test, but the test name shouldn't require that context and should match the assertion. That way the test makes sense to future users without having to look at the implementation. (as a side note this could also quite easily become obsolete as a test name if we renamed the param for example). A test name such as 'should not display edit actions within the premium payments table for PDC_READ users' might make more sense I think?

Reply:
Yep makes sense, just made that change thanks